lib/mongo/server_selector/selectable.rb

```diff
@@ -49,6 +53,8 @@ def initialize(options = nil)
         @options = options.freeze
         @tag_sets = (options[:tag_sets] || []).freeze
         @max_staleness = options[:max_staleness]
+        @hedge = options[:hedge]
```

Comment:
Can either all of these assignments be frozen or none? If all are frozen, the input arguments must be duplicated first since freeze mutates the object. E.g. `options.dup.freeze`, `(options[:tag_sets] || []).dup.freeze` .